addons/base_import/models/base_import.py

```diff
@@ -476,107 +554,208 @@ def _try_match_date_time(self, preview_values, options):
-            # FIXME: should match all translations & original
-            # TODO: use string distance (levenshtein? hamming?)
-            if header.lower() == field['name'].lower():
-                return [field]
-            if header.lower() == field['string'].lower():
-                # matching string are not reliable way because
-                # strings have no unique constraint
-                string_match = field
-            translated_header = IrTranslation._get_source('ir.model.fields,field_description', 'model', self.env.lang, header).lower()
-            if translated_header == field['string'].lower():
-                string_match = field
-        if string_match:
-            # this behavior is only applied if there is no matching field['name']
-            return [string_match]
+        if not fields_tree:
+            return {}
 
         if '/' not in header:
-            return []
+            # First, try exact match
+            IrTranslation = self.env['ir.translation']
+            for field in fields_tree:
+                # FIXME: should match all translations & original
+                string_match = None
+                if header.lower() == field['name'].lower():
+                    return {
+                        'field_path': [field['name']],
+                        'distance': 0
+                    }
+                field_string = field.get('string', '').lower()
+                if header.lower() == field_string:
+                    # matching string are not reliable way because
+                    # strings have no unique constraint
+                    string_match = field
+                translated_header = IrTranslation._get_source('ir.model.fields,field_description', 'model', self.env.lang, header).lower()
+                if translated_header == field_string:
+                    string_match = field
+                if string_match:
+                    # this behavior is only applied if there is no matching field['name']
+                    return {
+                        'field_path': [string_match['name']],
+                        'distance': 0
+                    }
+
+            # If no match found, try fuzzy match on most likely fields (based on header type)
+            # Filter out fields with types that does not match corresponding header types.
+            most_likely_fields = self._get_most_likely_fields(fields_tree, header_types)
+            if not most_likely_fields:
+                return {}
+
+            fields_matching_distance = {}
+            for field in most_likely_fields:
+                # use string distance for fuzzy match only on most likely field types
+                name_field_dist = Levenshtein.distance(header.lower(), field['name'].lower()) / max(len(field['name'].lower()), len(header.lower()))
+                string_field_dist = Levenshtein.distance(header.lower(), field_string) / max(len(field_string), len(header.lower()))
```

Comment:
It feels like your levenshtien comparison does not handle translations?
Or did I miss it?

Reply:
indeed. Will check with @odoongr if we will handle it (probably) or not